MLT-Lab-CSE6024_Lab-Assignment-2: Feature Engineering

Submitted By: 20MAI0001 - NIHARIKA MAITRA

Git Repo link :  https://github.com/Niharika-20-MAI-01/Winter-2021-MLTechniques-LabCSE6024-Niharika-20MAI01/blob/main/MLT-lab-CSE6024_20MAI0001_Feature_Engineering_Lab-Assignment-2.ipynb     

In [181]:
#Importing necessay Libraries and Packages
import pandas as pd
import numpy as np
import datetime

Performing Feature Engineering on crx.data dataset

In [182]:
#Loading the data to DataFrame using Pandas
data = pd.read_csv('crx.data', header = None)

1.	In the dataset crx.data, name the columns as col1, col2..col16

In [183]:
# Applying Variable Engineering or Feature Engineering to 
#Rename the Columns of crx.data dataset as col1,col2,...col16
varnames = ['Col'+ str(s) for s in range(1,17)]

In [184]:
#To add the new column names
data.columns = varnames

In [185]:
#To view the first 20 rows of dataset 
data.head(20)

Col1   Col2    Col3 Col4 Col5 Col6 Col7   Col8 Col9 Col10  Col11 Col12  \
0     b  30.83   0.000    u    g    w    v  1.250    t     t      1     f   
1     a  58.67   4.460    u    g    q    h  3.040    t     t      6     f   
2     a  24.50   0.500    u    g    q    h  1.500    t     f      0     f   
3     b  27.83   1.540    u    g    w    v  3.750    t     t      5     t   
4     b  20.17   5.625    u    g    w    v  1.710    t     f      0     f   
5     b  32.08   4.000    u    g    m    v  2.500    t     f      0     t   
6     b  33.17   1.040    u    g    r    h  6.500    t     f      0     t   
7     a  22.92  11.585    u    g   cc    v  0.040    t     f      0     f   
8     b  54.42   0.500    y    p    k    h  3.960    t     f      0     f   
9     b  42.50   4.915    y    p    w    v  3.165    t     f      0     t   
10    b  22.08   0.830    u    g    c    h  2.165    f     f      0     t   
11    b  29.92   1.835    u    g    c    h  4.335    t     f      0     f   
12    a  38.25   6.000    u    g    k    v  1.000    t     f      0     t   
13    b  48.08   6.040    u    g    k    v  0.040    f     f      0     f   
14    a  45.83  10.500    u    g    q    v  5.000    t     t      7     t   
15    b  36.67   4.415    y    p    k    v  0.250    t     t     10     t   
16    b  28.25   0.875    u    g    m    v  0.960    t     t      3     t   
17    a  23.25   5.875    u    g    q    v  3.170    t     t     10     f   
18    b  21.83   0.250    u    g    d    h  0.665    t     f      0     t   
19    a  19.17   8.585    u    g   cc    h  0.750    t     t      7     f   

   Col13  Col14  Col15 Col16  
0      g  00202      0     +  
1      g  00043    560     +  
2      g  00280    824     +  
3      g  00100      3     +  
4      s  00120      0     +  
5      g  00360      0     +  
6      g  00164  31285     +  
7      g  00080   1349     +  
8      g  00180    314     +  
9      g  00052   1442     +  
10     g  00128      0     +  
11     g  00260    200     +  
12     g  00000      0     +  
13     g  00000   2690     +  
14     g  00000      0     +  
15     g  00320      0     +  
16     g  00396      0     +  
17     g  00120    245     +  
18     g  00000      0     +  
19     g  00096      0     +

2.	Print the last 10 rows of data

In [186]:
##To view the last 10 rows of dataset 
data.tail(10)


Col1   Col2    Col3 Col4 Col5 Col6 Col7   Col8 Col9 Col10  Col11 Col12  \
680    b  19.50   0.290    u    g    k    v  0.290    f     f      0     f   
681    b  27.83   1.000    y    p    d    h  3.000    f     f      0     f   
682    b  17.08   3.290    u    g    i    v  0.335    f     f      0     t   
683    b  36.42   0.750    y    p    d    v  0.585    f     f      0     f   
684    b  40.58   3.290    u    g    m    v  3.500    f     f      0     t   
685    b  21.08  10.085    y    p    e    h  1.250    f     f      0     f   
686    a  22.67   0.750    u    g    c    v  2.000    f     t      2     t   
687    a  25.25  13.500    y    p   ff   ff  2.000    f     t      1     t   
688    b  17.92   0.205    u    g   aa    v  0.040    f     f      0     f   
689    b  35.00   3.375    u    g    c    h  8.290    f     f      0     t   

    Col13  Col14  Col15 Col16  
680     g  00280    364     -  
681     g  00176    537     -  
682     g  00140      2     -  
683     g  00240      3     -  
684     s  00400      0     -  
685     g  00260      0     -  
686     g  00200    394     -  
687     g  00200      1     -  
688     g  00280    750     -  
689     g  00000      0     -

3.	Replace the ‘?’ with Not-a-Number

In [187]:
#To replace ? with Not-an-Number (np.nan)
data = data.replace('?',np.nan)

In [188]:
data

Col1   Col2    Col3 Col4 Col5 Col6 Col7  Col8 Col9 Col10  Col11 Col12  \
0      b  30.83   0.000    u    g    w    v  1.25    t     t      1     f   
1      a  58.67   4.460    u    g    q    h  3.04    t     t      6     f   
2      a  24.50   0.500    u    g    q    h  1.50    t     f      0     f   
3      b  27.83   1.540    u    g    w    v  3.75    t     t      5     t   
4      b  20.17   5.625    u    g    w    v  1.71    t     f      0     f   
..   ...    ...     ...  ...  ...  ...  ...   ...  ...   ...    ...   ...   
685    b  21.08  10.085    y    p    e    h  1.25    f     f      0     f   
686    a  22.67   0.750    u    g    c    v  2.00    f     t      2     t   
687    a  25.25  13.500    y    p   ff   ff  2.00    f     t      1     t   
688    b  17.92   0.205    u    g   aa    v  0.04    f     f      0     f   
689    b  35.00   3.375    u    g    c    h  8.29    f     f      0     t   

    Col13  Col14  Col15 Col16  
0       g  00202      0     +  
1       g  00043    560     +  
2       g  00280    824     +  
3       g  00100      3     +  
4       s  00120      0     +  
..    ...    ...    ...   ...  
685     g  00260      0     -  
686     g  00200    394     -  
687     g  00200      1     -  
688     g  00280    750     -  
689     g  00000      0     -  

[690 rows x 16 columns]

4.	Comment on the datatype of variables

In [189]:
#To view the different Data Types of the variables present in the dataset
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 690 entries, 0 to 689
Data columns (total 16 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   Col1    678 non-null    object 
 1   Col2    678 non-null    object 
 2   Col3    690 non-null    float64
 3   Col4    684 non-null    object 
 4   Col5    684 non-null    object 
 5   Col6    681 non-null    object 
 6   Col7    681 non-null    object 
 7   Col8    690 non-null    float64
 8   Col9    690 non-null    object 
 9   Col10   690 non-null    object 
 10  Col11   690 non-null    int64  
 11  Col12   690 non-null    object 
 12  Col13   690 non-null    object 
 13  Col14   677 non-null    object 
 14  Col15   690 non-null    int64  
 15  Col16   690 non-null    object 
dtypes: float64(2), int64(2), object(12)
memory usage: 86.4+ KB


It can be observed that there are 3 different data types of the 
variables present in the crx.data dataset which are:-
 1) float64(2), 2) int64(2), 3) object(12)

In [190]:
#Recasting variables Col2 and Col14 to their correct data types
data['Col2'] = data['Col2'].astype('float')
data['Col14'] = data['Col14'].astype('float')

In [191]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 690 entries, 0 to 689
Data columns (total 16 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   Col1    678 non-null    object 
 1   Col2    678 non-null    float64
 2   Col3    690 non-null    float64
 3   Col4    684 non-null    object 
 4   Col5    684 non-null    object 
 5   Col6    681 non-null    object 
 6   Col7    681 non-null    object 
 7   Col8    690 non-null    float64
 8   Col9    690 non-null    object 
 9   Col10   690 non-null    object 
 10  Col11   690 non-null    int64  
 11  Col12   690 non-null    object 
 12  Col13   690 non-null    object 
 13  Col14   677 non-null    float64
 14  Col15   690 non-null    int64  
 15  Col16   690 non-null    object 
dtypes: float64(4), int64(2), object(10)
memory usage: 86.4+ KB


5.	The col16 has + and -, replace them ‘P’ and ‘N’ respectively

In [192]:
#To encode Col16 values from + and - to 'P' and 'N' respectively
data['Col16'] = data['Col16'].map({'+':'P', '-':'N'})

In [193]:
data

Col1   Col2    Col3 Col4 Col5 Col6 Col7  Col8 Col9 Col10  Col11 Col12  \
0      b  30.83   0.000    u    g    w    v  1.25    t     t      1     f   
1      a  58.67   4.460    u    g    q    h  3.04    t     t      6     f   
2      a  24.50   0.500    u    g    q    h  1.50    t     f      0     f   
3      b  27.83   1.540    u    g    w    v  3.75    t     t      5     t   
4      b  20.17   5.625    u    g    w    v  1.71    t     f      0     f   
..   ...    ...     ...  ...  ...  ...  ...   ...  ...   ...    ...   ...   
685    b  21.08  10.085    y    p    e    h  1.25    f     f      0     f   
686    a  22.67   0.750    u    g    c    v  2.00    f     t      2     t   
687    a  25.25  13.500    y    p   ff   ff  2.00    f     t      1     t   
688    b  17.92   0.205    u    g   aa    v  0.04    f     f      0     f   
689    b  35.00   3.375    u    g    c    h  8.29    f     f      0     t   

    Col13  Col14  Col15 Col16  
0       g  202.0      0     P  
1       g   43.0    560     P  
2       g  280.0    824     P  
3       g  100.0      3     P  
4       s  120.0      0     P  
..    ...    ...    ...   ...  
685     g  260.0      0     N  
686     g  200.0    394     N  
687     g  200.0      1     N  
688     g  280.0    750     N  
689     g    0.0      0     N  

[690 rows x 16 columns]

6.	Find and display the number of  variables of type ‘Object’

In [194]:
#To find and display the Number of Object type variables present in the dataset
obj_columns = [i for i in data.columns if data[i].dtypes == 'O']
print(obj_columns)


['Col1', 'Col4', 'Col5', 'Col6', 'Col7', 'Col9', 'Col10', 'Col12', 'Col13', 'Col16']


In [195]:
data[obj_columns]

Col1 Col4 Col5 Col6 Col7 Col9 Col10 Col12 Col13 Col16
0      b    u    g    w    v    t     t     f     g     P
1      a    u    g    q    h    t     t     f     g     P
2      a    u    g    q    h    t     f     f     g     P
3      b    u    g    w    v    t     t     t     g     P
4      b    u    g    w    v    t     f     f     s     P
..   ...  ...  ...  ...  ...  ...   ...   ...   ...   ...
685    b    y    p    e    h    f     f     f     g     N
686    a    u    g    c    v    f     t     t     g     N
687    a    y    p   ff   ff    f     t     t     g     N
688    b    u    g   aa    v    f     f     f     g     N
689    b    u    g    c    h    f     f     t     g     N

[690 rows x 10 columns]

Performing Feature Engineering on loan.csv dataset

In [196]:
#Loading the data to DataFrame using Pandas
data1 = pd.read_csv('loan.csv')

In [197]:
data1

customer_id  disbursed_amount  interest market     employment  \
0               0           23201.5   15.4840      C        Teacher   
1               1            7425.0   11.2032      B     Accountant   
2               2           11150.0    8.5100      A   Statistician   
3               3            7600.0    5.8656      A          Other   
4               4           31960.0   18.7392      E     Bus driver   
...           ...               ...       ...    ...            ...   
9995         9995           23750.0   11.0019      B  Civil Servant   
9996         9996           11880.0   10.4923      B  Civil Servant   
9997         9997           19950.0    8.4364      B     Accountant   
9998         9998            4850.0   13.6409      C     Bus driver   
9999         9999           25584.0   17.1456      D     Accountant   

     time_employed householder    income date_issued  target  \
0        <=5 years        RENT   84600.0  2013-06-11       0   
1        <=5 years       OWNER  102000.0  2014-05-08       0   
2        <=5 years        RENT   69840.0  2013-10-26       0   
3        <=5 years        RENT  100386.0  2015-08-20       0   
4         >5 years        RENT   95040.0  2014-07-22       0   
...            ...         ...       ...         ...     ...   
9995     <=5 years    MORTGAGE   98000.0  2010-01-14       0   
9996     <=5 years        RENT   36270.0  2015-03-20       0   
9997      >5 years       OWNER   52250.0  2015-04-03       0   
9998      >5 years        RENT   80040.0  2014-11-23       0   
9999      >5 years    MORTGAGE   72000.0  2015-01-19       0   

            loan_purpose  number_open_accounts date_last_payment  \
0     Debt consolidation                   4.0        2016-01-14   
1           Car purchase                  13.0        2016-01-25   
2     Debt consolidation                   8.0        2014-09-26   
3     Debt consolidation                  20.0        2016-01-26   
4     Debt consolidation                  14.0        2016-01-11   
...                  ...                   ...               ...   
9995               Other                  12.0        2010-11-03   
9996  Debt consolidation                  17.0        2016-01-26   
9997  Debt consolidation                   9.0        2015-12-12   
9998  Debt consolidation                  26.0        2016-01-01   
9999  Debt consolidation                   5.0        2016-01-16   

      number_credit_lines_12  
0                        NaN  
1                        NaN  
2                        NaN  
3                        NaN  
4                        NaN  
...                      ...  
9995                     NaN  
9996                     NaN  
9997                     NaN  
9998                     NaN  
9999                     NaN  

[10000 rows x 14 columns]

In [198]:
data1.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10000 entries, 0 to 9999
Data columns (total 14 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   customer_id             10000 non-null  int64  
 1   disbursed_amount        10000 non-null  float64
 2   interest                10000 non-null  float64
 3   market                  10000 non-null  object 
 4   employment              9389 non-null   object 
 5   time_employed           9471 non-null   object 
 6   householder             10000 non-null  object 
 7   income                  10000 non-null  float64
 8   date_issued             10000 non-null  object 
 9   target                  10000 non-null  int64  
 10  loan_purpose            10000 non-null  object 
 11  number_open_accounts    10000 non-null  float64
 12  date_last_payment       10000 non-null  object 
 13  number_credit_lines_12  238 non-null    float64
dtypes: float64(5), int64(2), object(7)
memo

1.	Display the mean of any two variables with continuous values

In [199]:
#To display the Statistical Parameter Mean of the variable 'disbursed_amount' 
#which consists of  continuous values
m1=data1['disbursed_amount'].mean()
print('Mean of column disbursed_amount = '+str(m1))

Mean of column disbursed_amount = 14132.2755


In [200]:
#To display the Statistical Parameter Mean of the variable 'interest' 
#which consists of  continuous values 
m3=data1['interest'].mean()
print('Mean of column interest = '+str(m3))

Mean of column interest = 12.678819440000039


2.	Print the number of discrete variables

In [201]:
#Recasting variable 'number_open_accounts' to its correct data type
data1['number_open_accounts'] = data1['number_open_accounts'].astype('int')

In [202]:
num = ['int16', 'int32', 'int64']
df1 = data1.select_dtypes(include=num)

In [203]:
#To print the number of discrete variables present in the dataset
df1

customer_id  target  number_open_accounts
0               0       0                     4
1               1       0                    13
2               2       0                     8
3               3       0                    20
4               4       0                    14
...           ...     ...                   ...
9995         9995       0                    12
9996         9996       0                    17
9997         9997       0                     9
9998         9998       0                    26
9999         9999       0                     5

[10000 rows x 3 columns]

It can be observed that there are 3 Discrete Variables present in the loan.csv dataset which are:-
1) 'customer_id'  2) 'target'  3) 'number_open_accounts'     

3.	Display the unique values of two variables with discrete values

In [204]:
#To display the unique values of the discrete variable 'number_open_accounts'
data1['number_open_accounts'].unique()

array([ 4, 13,  8, 20, 14,  5,  9, 18, 16, 17, 12, 15,  6, 10, 11,  7, 21,
       19, 26,  2, 22, 27, 23, 25, 24, 28,  3, 30, 41, 32, 33, 31, 29, 37,
       49, 34, 35, 38,  1, 36, 42, 47, 40, 44, 43])

In [205]:
#To display the unique values of the discrete variable 'target'
data1['target'].unique()

array([0, 1], dtype=int64)

In [206]:
#To display the unique values of the discrete variable 'customer_id'
data1['customer_id'].unique()

array([   0,    1,    2, ..., 9997, 9998, 9999], dtype=int64)


4.	Display the Month with most of loans issued date

In [207]:
#To view the unique values of the variable 'date_issued'
data1['date_issued'].unique()


array(['2013-06-11', '2014-05-08', '2013-10-26', ..., '2010-08-22',
       '2009-06-02', '2010-01-14'], dtype=object)

In [208]:
#Converting and storing 'date_issued' variable in date-time format as 'date_issued_dt' 
#variable appended in the dataset
data1['date_issued_dt'] = pd.to_datetime(data1['date_issued'])
data1.head()

customer_id  disbursed_amount  interest market    employment time_employed  \
0            0           23201.5   15.4840      C       Teacher     <=5 years   
1            1            7425.0   11.2032      B    Accountant     <=5 years   
2            2           11150.0    8.5100      A  Statistician     <=5 years   
3            3            7600.0    5.8656      A         Other     <=5 years   
4            4           31960.0   18.7392      E    Bus driver      >5 years   

  householder    income date_issued  target        loan_purpose  \
0        RENT   84600.0  2013-06-11       0  Debt consolidation   
1       OWNER  102000.0  2014-05-08       0        Car purchase   
2        RENT   69840.0  2013-10-26       0  Debt consolidation   
3        RENT  100386.0  2015-08-20       0  Debt consolidation   
4        RENT   95040.0  2014-07-22       0  Debt consolidation   

   number_open_accounts date_last_payment  number_credit_lines_12  \
0                     4        2016-01-14                     NaN   
1                    13        2016-01-25                     NaN   
2                     8        2014-09-26                     NaN   
3                    20        2016-01-26                     NaN   
4                    14        2016-01-11                     NaN   

  date_issued_dt  
0     2013-06-11  
1     2014-05-08  
2     2013-10-26  
3     2015-08-20  
4     2014-07-22

In [209]:
#To extract the Months from the date-time values of 'date_issued_dt' Variable
data1['month'] = data1['date_issued_dt'].dt.month


In [210]:
#To display the number of occurrence of each unique value of the variable 'month'
data1['month'].value_counts()

10    1277
7     1066
11    1017
12     882
8      852
4      816
5      749
9      734
6      700
1      700
3      623
2      584
Name: month, dtype: int64

In [211]:
print(data1.groupby('month').count())

       customer_id  disbursed_amount  interest  market  employment  \
month                                                                
1              700               700       700     700         655   
2              584               584       584     584         536   
3              623               623       623     623         588   
4              816               816       816     816         779   
5              749               749       749     749         704   
6              700               700       700     700         665   
7             1066              1066      1066    1066        1006   
8              852               852       852     852         792   
9              734               734       734     734         687   
10            1277              1277      1277    1277        1212   
11            1017              1017      1017    1017         944   
12             882               882       882     882         821   

       time_employe

It can be observed that the 10 th Month i.e, the Month of October has the highest/most numbers of Loans issued date


5.	Display the count of ‘Teacher’ who are ‘Owners’

In [212]:
#To display the number of occurrence of each value of the variable 'employment'
data1['employment'].value_counts()

Civil Servant         884
Nurse                 880
Bus driver            874
Other                 863
Teacher               861
Accountant            852
Statistician          852
Secretary             846
Dentist               844
Taxi driver           825
Software developer    808
Name: employment, dtype: int64

In [213]:
#To display the number of occurrence of each value of the variable 'householder'
data1['householder'].value_counts()

MORTGAGE    4957
RENT        4055
OWNER        988
Name: householder, dtype: int64

In [214]:
#To display the frequency/count of all the combinations of the values of the two
#variables 'employment' and 'householder' of the dataset
data1['Emp_Hous'] = data1['employment'].str.cat(data1['householder'],sep=" ")
print(data1['Emp_Hous'].value_counts())

Nurse MORTGAGE                 444
Accountant MORTGAGE            440
Secretary MORTGAGE             432
Taxi driver MORTGAGE           429
Statistician MORTGAGE          426
Other MORTGAGE                 425
Civil Servant MORTGAGE         424
Teacher MORTGAGE               421
Bus driver MORTGAGE            418
Dentist MORTGAGE               418
Software developer MORTGAGE    417
Civil Servant RENT             371
Teacher RENT                   371
Bus driver RENT                360
Nurse RENT                     358
Dentist RENT                   355
Secretary RENT                 355
Other RENT                     353
Statistician RENT              342
Accountant RENT                322
Taxi driver RENT               316
Software developer RENT        315
Bus driver OWNER                96
Accountant OWNER                90
Civil Servant OWNER             89
Other OWNER                     85
Statistician OWNER              84
Taxi driver OWNER               80
Nurse OWNER         

In [215]:
#Selecting all the rows of the dataset having 'employment' variable value eqaul to 'TEACHER'
data2 = data1[data1['employment']=='Teacher']

In [216]:
#Selecting all the rows of the dataset having 'householder' variable value eqaul to 'OWNER'
data3 = data2[data2['householder']=='OWNER']

In [217]:
#To  Display the count of ‘Teacher’ who are ‘Owners’
data3['Teach_Own'] = data3['householder'].str.cat(data3['employment'],sep=" ")
print(data3['Teach_Own'].value_counts())

OWNER Teacher    69
Name: Teach_Own, dtype: int64


<ipython-input-217-ebf08aafce19>:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data3['Teach_Own'] = data3['householder'].str.cat(data3['employment'],sep=" ")


It can be observed that the count of 'Teacher' who are 'Owners' is 69


6.	Display the ‘Employment’ of customers who mostly ‘Rent’


In [218]:
#To display the number of occurrence of each value of the variable 'employment'
data1['employment'].value_counts()

Civil Servant         884
Nurse                 880
Bus driver            874
Other                 863
Teacher               861
Accountant            852
Statistician          852
Secretary             846
Dentist               844
Taxi driver           825
Software developer    808
Name: employment, dtype: int64

In [219]:
#To display the number of occurrence of each value of the variable 'householder'
data1['householder'].value_counts()

MORTGAGE    4957
RENT        4055
OWNER        988
Name: householder, dtype: int64

In [220]:
#To display the frequency/count of all the combinations of the values of the two
#variables 'employment' and 'householder' of the dataset
data1['Emp_Hous'] = data1['employment'].str.cat(data1['householder'],sep=" ")
print(data1['Emp_Hous'].value_counts())

Nurse MORTGAGE                 444
Accountant MORTGAGE            440
Secretary MORTGAGE             432
Taxi driver MORTGAGE           429
Statistician MORTGAGE          426
Other MORTGAGE                 425
Civil Servant MORTGAGE         424
Teacher MORTGAGE               421
Bus driver MORTGAGE            418
Dentist MORTGAGE               418
Software developer MORTGAGE    417
Civil Servant RENT             371
Teacher RENT                   371
Bus driver RENT                360
Nurse RENT                     358
Dentist RENT                   355
Secretary RENT                 355
Other RENT                     353
Statistician RENT              342
Accountant RENT                322
Taxi driver RENT               316
Software developer RENT        315
Bus driver OWNER                96
Accountant OWNER                90
Civil Servant OWNER             89
Other OWNER                     85
Statistician OWNER              84
Taxi driver OWNER               80
Nurse OWNER         

In [221]:
#Selecting all the rows of the dataset having 'householder' variable value eqaul to 'RENT'
data4 = data1[data1['householder']=='RENT']


In [222]:
#To display the frequency/count of all the combinations of the values of the two
#variables 'employment' and 'householder' equals to 'RENT' of the dataset

data4['Rent_Emp'] = data4['householder'].str.cat(data4['employment'],sep=" ")

c = data4['Rent_Emp'].value_counts()
print(c)


RENT Civil Servant         371
RENT Teacher               371
RENT Bus driver            360
RENT Nurse                 358
RENT Secretary             355
RENT Dentist               355
RENT Other                 353
RENT Statistician          342
RENT Accountant            322
RENT Taxi driver           316
RENT Software developer    315
Name: Rent_Emp, dtype: int64


<ipython-input-222-118e505d17d0>:4: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data4['Rent_Emp'] = data4['householder'].str.cat(data4['employment'],sep=" ")


In [223]:
c.max()

371

In [224]:
emp = ['Teacher', 'Civil Servant']
data5 = data4[data4['employment'].isin(emp) ]

In [225]:
# To Display the ‘Employment’ of customers who mostly ‘Rent’

data5['Emp-max_rent']= data5['householder'].str.cat(data5['employment'],sep=" ")
print(data5['Emp-max_rent'].value_counts())

RENT Civil Servant    371
RENT Teacher          371
Name: Emp-max_rent, dtype: int64


<ipython-input-225-e49caf6fc9e0>:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data5['Emp-max_rent']= data5['householder'].str.cat(data5['employment'],sep=" ")


It can be observed that the maximum count of 'Employment'  who are 'Owners' is 371 and 
the employment type with this maximum value are 'Teacher' and  'Civil Servant'